packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -47,7 +47,7 @@ abstract class DevFSContent {
   Stream<List<int>> contentsAsStream();
 
   Stream<List<int>> contentsAsCompressedStream() {
-    return contentsAsStream().cast<List<int>>().transform<List<int>>(gzip.encoder);
+    return globals.os.gzipLevel1Stream(contentsAsStream());
```

Comment:
Thinking about how this API will evolve in a world without globals. Either:

1. every DevFSContent needs a reference to `OperatingSystemUtils`
2. `contentsAsCompressedStream` accepts an `OperatingSystemUtils` or a `  Stream<List<int>> Function(Stream<List<int>>)` torn off from one.

The latter case sounds more appealing to me, since it simplifies the construction of the DevFSContent classes and reduces the dependencies. In that case, instead of mocking the `OperatingSystemUtils` for the current test, you might need to do something different.

Reply:
Done